tokenmagic/module/tokenmagic.js

```diff
@@ -2013,21 +2013,21 @@ Hooks.on("preCreateMeasuredTemplate", (document, data, options, user) => {
       if (persist) tmfxFiltersData = newFilters;
     }
   } else {
-    data.tmfxPreset = emptyPreset;
+    document.tmfxPreset = emptyPreset;
   }
 
-  if (!hasOpacity) data.tmfxTextureAlpha = 1;
-  if (!hasTint) data.tmfxTint = null;
+  if (!hasOpacity) document.tmfxTextureAlpha = 1;
+  if (!hasTint) document.tmfxTint = null;
 
   let tmfxFlags = {
     templateData: {
-      opacity: data.tmfxTextureAlpha,
-      tint: data.tmfxTint
+      opacity: document.tmfxTextureAlpha,
+      tint: document.tmfxTint
     },
     filters: tmfxFiltersData,
     options: null
   };
-  document.data.update({ flags: { tokenmagic: tmfxFlags } });
+  //document.update({ flags: { tokenmagic: tmfxFlags } });

```

Comment:
I can't find anything that suggests what to do with the required `_id` that doesn't exist yet as this hook is called before the template exists.

Reply:
Temporarily fixed this by changed it to the `create` event rather than the `preCreate` - far from ideal but moves us closer to V10 and can be addressed later.